packages/core/src/app/shipping/Shipping.tsx

```diff
@@ -191,15 +193,18 @@ class Shipping extends Component<ShippingProps & WithCheckoutShippingProps, Ship
             billingAddress,
             methodId,
             emitAnalyticsEvent,
+            isShippingDetailsEntered,
         } = this.props;
 
         const updatedShippingAddress = addressValues && mapAddressFromFormValues(addressValues);
         const promises: Array<Promise<CheckoutSelectors>> = [];
         const hasRemoteBilling = this.hasRemoteBilling(methodId);
-
-        emitAnalyticsEvent("Shipping method step complete");
+        if (!isShippingDetailsEntered) {
+            emitAnalyticsEvent(GuestCheckoutEvents.ShippingEntered);
```

Comment:
Why are we emitting the ShippingEntered event from `handleSingleShippingSubmit`? Why not in  componentDidMount or render?

Reply:
From the variable name, it looks like we want to emit an event when we enter the shipping section. This code path gets called when a shopper has already entered their shipping details so its a bit strange

Reply:
technically this event should be called when the shipping form is fully filled and shipping options start to load: https://docs.google.com/spreadsheets/d/11BNt3pSiQZq34zXMOfojOIs7RjbVm-mD8QA0s255viE/edit#gid=0

Reply:
Oh i see, thanks for clarifying